.circleci/config.yml

```diff
@@ -154,18 +162,36 @@ commands:
           paths:
             - ".local-*"
 
-  prepare-python:
+  prepare-pip-cache:
     parameters:
       py-version:
         <<: *py-version-template
     steps:
-      - attach-and-link:
-          py-version: << parameters.py-version >>
-      - conditional-skip
-      - config-path
       - restore_cache:
+          keys:
+          - pip-cache-{{ arch }}-<< parameters.py-version >>
+          - pip-cache-{{ arch }}-
+          - pip-cache-
+      - run:
+          name: Downloading dependencies
+          command: pip download --dest /tmp -c constraints-dev.txt -r requirements-circleci.txt
+      - save_cache:
           key: pip-cache-{{ arch }}-<< parameters.py-version >>
+          paths:
+          - "~/.cache/pip"
+
+  install-raiden:
```

Comment:
My idea is to add another command that *does not install raiden*, because our `setup.py` currently opens the `constraints.txt` file and makes everything a direct dependency of raiden. Instead I will run `pip install -r requirements.txt -c constraints.txt` and assert the `pip freeze` and existing  `constraints.txt` are equal

Reply:
Hm I don't think that will work. 
Dominik told me about the readline problem. That is easily fixed by making it platform dependant in constraints but then it won't exactly match the output of freeze.

Reply:
There was no readline problem. His system was just missing ncurses.

Reply:
> That is easily fixed by making it platform dependant in constraints but then it won't exactly match the output of freeze

Yeah, my idea is to have separate constraints files for each OS